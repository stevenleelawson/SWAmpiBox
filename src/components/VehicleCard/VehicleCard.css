.vehicle-name {
  color: white;
}

.vehicle {
  border: 3px solid #fff;
  border-radius: 5px;
  height: 15rem;
  width: 15rem;
  margin: 5rem;
}
