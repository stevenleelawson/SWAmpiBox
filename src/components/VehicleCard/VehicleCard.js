import React from 'react';
import './VehicleCard.css'

const VehicleCard = ({name, model, vehicleClass, passengers, toggleFavorites}) => {
  return (
    <div className='vehicle'>
      <h1 className='vehicle-name'>{name}</h1>
      <p className='vehicle-model'>model: {model}</p>
      <p className='vehicle-class'>class: {vehicleClass}</p>
      <p className='vehicle-passenger'>passengers: {passengers}</p>
      <button className='button' onClick={() => toggleFavorites({name, model, vehicleClass, passengers, category:'vehicles'})}>Favorite</button>
    </div>
  )
}

export default VehicleCard;
