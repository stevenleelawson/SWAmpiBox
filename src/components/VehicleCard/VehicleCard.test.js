import React from 'react';
import { shallow } from 'enzyme';
import VehicleCard from './VehicleCard';

describe('VehicleCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<VehicleCard />);
    expect(wrapper).toMatchSnapshot();
  })
})
