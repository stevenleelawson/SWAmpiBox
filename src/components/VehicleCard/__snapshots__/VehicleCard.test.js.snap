// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="vehicle-name"
/>,
      "className": "vehicle",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "vehicle-name",
      },
      "ref": null,
      "rendered": null,
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="vehicle-name"
/>,
        "className": "vehicle",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "vehicle-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
